torch/_dynamo/variables/builtin.py

```diff
@@ -1106,6 +1106,11 @@ def _comparison(self, tx, left, right):
         def _unimplemented():
             unimplemented(f"comparison {typestr(left)} {op} {typestr(right)}")
 
+        if left.is_python_constant() and right.is_python_constant():
+            return ConstantVariable(
+                op(left.as_python_constant(), right.as_python_constant())
+            )
+
```

Comment:
I think ```_comparison``` is used to handle non constant fold through cases, like the comments you added at ```call_or_```, we only get here if constant fold is not valid. But right now, you are trying to add constant fold logics inside of these functions to make them handle both constant through and not logics. I think this will make us even hard to maintain the codebase. I'm ok with this change now, but we can revisit later if we have more use cases, to keep these function semantics more consistent.